### PR TITLE
Add dev guide for new controls, allow better view of notebook. 

### PR DESCRIPTION
**Pull request by gbowlin**
# Overview

## Description of changes
Adds a section to the dev guide to show off making your own custom control. 
image

Updates docstrings so they appear in sphinx docs as code blocks
image

Allows a larger than 700px view for larger screen support. 
image

Fixes an issue with `sm.cohort_list()` running off the side in the Sphinx docs:
image

## Author Checklist
- [ ] Linting passes; run early with [pre-commit hook](https://pre-commit.com/#install).
- [ ] Tests added for new code and issue being fixed.
- [ ] Added type annotations and full numpy-style docstrings for new methods.
- [ ] Draft your news fragment in new `changelog/ISSUE.TYPE.rst` files; see [changelog/README.md](https://github.com/epic-open-source/seismometer/blob/main/changelog/README.md).

### CODE REVIEW

File: docs/user_guide/index.rst [hunk 1]
```diff
@@ -533,3 +533,66 @@ Then, follow the pattern of normal startup but specify your function in the :py:
 
 The benefit of this approach over manipulating the frame later is that the Seismogram can be considered frozen.
 Among other things, this means any Seismograph notebooks cells do not have a dependence on order and can be run multiple times.
+
+
+Create Custom Visualizations
```
Comment (gbowlin) [inline docs/user_guide/index.rst:538]: @firste - mostly looking for feedback on this blurb

File: docs/reference/seismometer.rst [hunk 2]
```diff
@@ -38,3 +38,20 @@ Public API
...
+.. currentmodule:: seismometer.controls
+
+.. autosummary::
+   :toctree: api/
+
+
+   explore.ExplorationWidget
+   explore.ExplorationModelSubgroupEvaluationWidget
+   explore.ExplorationCohortSubclassEvaluationWidget
+   explore.ExplorationCohortOutcomeInterventionEvaluationWidget
```
Comment (firste) [inline docs/reference/seismometer.rst:57]: These both have the same description. Can you update the intervention eval version to include the word intervention?

**Review by firste — CHANGES_REQUESTED** [review]

File: docs/user_guide/index.rst (hunk 1 above)
Comment (firste) [reply docs/user_guide/index.rst:538]: Looks good to me.

File: docs/user_guide/index.rst [hunk 3]
```diff
@@ -533,3 +533,66 @@ Then, follow the pattern of normal startup but specify your function in the :py:
...
+standardized controls. The :py:mod:`seismometer.controls.explore` module contains several ``Exploration*``
+widgets you can use for housing custom visualizations, see :ref:`custom-visualization-controls`. 
+
+.. image:: media/custom_plot_control.png
+   :alt: A custom control, allowing a user to select a cohort and display a heatmap restricted to that cohort.
+   :width: 4.5in
+
+To add your own custom visualization, you need a function that takes the same signature as the Exploration widget, and
+it should return a displayable object. If using matplotlib, you can use the :py:func:`~seismometer.plot.mpl.decorators.render_as_svg`
+to convert the plot to an SVG, for the control to display.
```
Comment (diehlbw) [inline docs/user_guide/index.rst:551]:
```suggestion
decorator to convert the plot to an SVG, for the control to display.
```

File: docs/user_guide/index.rst [hunk 4]
```diff
@@ -533,3 +533,66 @@ Then, follow the pattern of normal startup but specify your function in the :py:
...
+This will close the plot/figure after saving to prevent the plot from displaying twice.
+
+The following example shows how to create a visualization above.
```
Comment (diehlbw) [inline docs/user_guide/index.rst:554]:
```suggestion
The following example shows how to create the visualization above. 
```

File: docs/user_guide/index.rst [hunk 5]
```diff
@@ -533,3 +533,66 @@ Then, follow the pattern of normal startup but specify your function in the :py:
...
+   import matplotlib.pyplot as plt
+
+   # Control allowing users to specify a score, target, threshold, and cohort. 
+   from seismometer.controls.explore import ExplorationModelSubgroupEvaluationWidget 
+   # Converts matplotlib figure to SVG for display within the control's output
+   from seismometer.plot.mpl.decorators import render_as_svg
+   # Filter our data based on a specified cohort 
+   from seismometer.data.filter import FilterRule 
+
+   @render_as_svg
```
Comment (diehlbw) [inline docs/user_guide/index.rst:568]: I think it'd be good to explicitly state the signature is determined by the control being used.  Maybe even calling out that ignoring arguments is ok, if it's ok to have controls that don't drive content updates.

File: docs/user_guide/index.rst [hunk 6]
```diff
@@ -533,3 +533,66 @@ Then, follow the pattern of normal startup but specify your function in the :py:
...
+         score_col: str,  # the model output column (score)
+         thresholds: tuple[float], # a list of thresholds to consider
+         *,
+         per_context: bool # if a plot groups scores by context
+         ) -> plt.Figure:
+      xcol = "age"
+      ycol = "num_procedures"
+      hue = score_col
+
+      sg = sm.Seismogram()
```
Comment (diehlbw) [inline docs/user_guide/index.rst:581]: It's probably worth while to split the cohort_filter.filter() and pandas slicing (line~584), that way we can call out these "three" lines (582-584) are going to be useful in their plot exactly.

File: docs/user_guide/index.rst [hunk 7]
```diff
@@ -533,3 +533,66 @@ Then, follow the pattern of normal startup but specify your function in the :py:
...
+   from seismometer.data.filter import FilterRule 
+
+   @render_as_svg
+   def plot_heat_map(
+         cohort_dict: dict[str,tuple], # cohort columns and allowable values
+         target_col: str, # the model target column
+         score_col: str,  # the model output column (score)
+         thresholds: tuple[float], # a list of thresholds to consider
+         *,
+         per_context: bool # if a plot groups scores by context
```
Comment (diehlbw) [inline docs/user_guide/index.rst:575]: probably more complexity than we want, but would it be useful to extend this to show accessing sg.event_aggregation_method(target_col) when wanting per_context?

Or maybe on the other side, calling out this as an example of an ignored param(checkbox) that is still visible in the control regardless of use

Comment (gbowlin) [reply docs/user_guide/index.rst:575]: will add an "ignored in this example" comment

File: docs/reference/seismometer.rst (hunk 2 above)
Comment (gbowlin) [reply docs/reference/seismometer.rst:57]: FINC

File: docs/user_guide/index.rst (hunk 4 above)
Comment (gbowlin) [reply docs/user_guide/index.rst:554]: FINC

File: docs/user_guide/index.rst (hunk 5 above)
Comment (gbowlin) [reply docs/user_guide/index.rst:568]: FINC

File: docs/user_guide/index.rst (hunk 6 above)
Comment (gbowlin) [reply docs/user_guide/index.rst:581]: FINC

**Review by diehlbw — APPROVED** [review]

File: src/seismometer/controls/explore.py [hunk 8]
```diff
@@ -344,7 +344,7 @@ def group_scores(self) -> bool:
 
 
 class ModelInterventionOptionsWidget(VBox, ValueWidget):
-    value = traitlets.Dict(help="The selected values for the slider list")
+    value = traitlets.Dict(help="The selected values for the intevetion options")
```
Comment (firste) [inline src/seismometer/controls/explore.py:347]:
```suggestion
    value = traitlets.Dict(help="The selected values for the intervention options")
```

**Review by firste — CHANGES_REQUESTED** [review]

**Review by firste — APPROVED** [review]

**Pull request merged**